ext/stackprof/stackprof.c

```diff
@@ -177,6 +193,23 @@ stackprof_start(int argc, VALUE *argv, VALUE self)
 	    raw = 1;
 	if (rb_hash_lookup2(opts, sym_aggregate, Qundef) == Qfalse)
 	    aggregate = 0;
+
+
+	if (RTEST(rb_hash_aref(opts, sym_tag_source))) {
+	    tag_source = rb_hash_aref(opts, sym_tag_source);
+	    if (!RB_TYPE_P(tag_source, T_SYMBOL))
+		rb_raise(rb_eArgError, "tag source should be the symbol of a fiber local variable to check for tags. Tags should be keyed by a symbol, and the value should be a string or a symbol");
+	} else {
+	    tag_source = sym__stackprof_tags;
+	}
+
+	if (RTEST(rb_hash_aref(opts, sym_tags))) {
+	    tags = rb_hash_aref(opts, sym_tags);
+	    if (!RB_TYPE_P(tags, T_ARRAY))
```

Comment:
TODO check if the array contains sym_thread_id, and if it does, remove it, set a boolean.